cmake/FindCatch.cmake

```diff
@@ -29,7 +29,7 @@ endfunction()
 
 find_path(
   CATCH_INCLUDE_DIR
-  PATH_SUFFIXES single_include include catch catch2
+  PATH_SUFFIXES single_include include catch catch2 include/catch2
```

Comment:
Re commit message:
> Spack installs the headers in include/catch2.

If I'm interpreting this right that sounds like a Spack bug.  Is there an upstream issue or anything?

Reply:
Is it a bug? Isn't it just fine to install in `include/catch2` so headers are included as `catch2/*.hpp`?

Reply:
OK, I guess it's CMake stupid, then.  (That's the completely standard path and how I've always had it installed, so I assumed it was installing to a nested `include` or something.)